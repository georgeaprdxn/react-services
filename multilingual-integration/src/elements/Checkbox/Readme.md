Checkbox

```jsx inside Markdown
import mockData from './__test__/Checkbox.mock.json'
const [isChecked, setIsChecked] = React.useState(false)
const changeHandler = e => setIsChecked(e.target.checked)
;<Checkbox
	label={mockData.label}
	name={mockData.name}
	id={mockData.id}
	isChecked={isChecked}
	handleChange={e => changeHandler(e)}
/>
```
