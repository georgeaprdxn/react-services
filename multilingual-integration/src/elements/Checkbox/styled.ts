import styled from 'styled-components'

const StyledCheckbox = styled.div``

const Input = styled.input``

const Label = styled.label``

export { StyledCheckbox, Input, Label }
