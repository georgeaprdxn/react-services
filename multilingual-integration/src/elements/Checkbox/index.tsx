import React, { FunctionComponent } from 'react'

import { StyledCheckbox, Input, Label } from './styled'

type Props = {
	/**
	 * label to the checkbox
	 */
	label: string
	/**
	 * Name of the checkbox
	 */
	name: string
	/**
	 * unique identifier to the checkbox
	 */
	id: string
	/**
	 * handles change event
	 */
	handleChange?: (e: React.ChangeEvent<HTMLInputElement>) => void
	/**
	 * state of checkbox
	 */
	isChecked: boolean
}

const Checkbox: FunctionComponent<Props> = ({
	label,
	name,
	id,
	handleChange,
	isChecked,
}) => {
	return (
		<StyledCheckbox>
			<Input
				type="checkbox"
				name={name}
				id={id}
				onChange={handleChange}
				checked={isChecked}
			/>
			<Label htmlFor={id} title={label}>
				{label}
			</Label>
		</StyledCheckbox>
	)
}

Checkbox.defaultProps = {
	handleChange: () => undefined,
}

export default Checkbox
