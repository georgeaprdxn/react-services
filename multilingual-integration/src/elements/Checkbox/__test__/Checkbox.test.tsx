import React from 'react'
import { shallow } from 'enzyme'

import Checkbox from '../index'
import mockCheckboxData from './Checkbox.mock.json'

describe('Checkbox element tests', () => {
	let checkboxWrapper: any = null
	const handleChangeStub = jest.fn()
	beforeEach(() => {
		checkboxWrapper = shallow(
			<Checkbox
				label={mockCheckboxData.label}
				name={mockCheckboxData.name}
				id={mockCheckboxData.id}
				isChecked={mockCheckboxData.isChecked}
				handleChange={handleChangeStub}
			/>
		)
	})

	it('should render a checkbox', () => {
		expect(
			checkboxWrapper.find(`#${mockCheckboxData.id}`).prop('type')
		).toEqual('checkbox')
	})

	it('validates all props', () => {
		expect(
			checkboxWrapper.find(`#${mockCheckboxData.id}`).prop('name')
		).toEqual(mockCheckboxData.name)

		expect(checkboxWrapper.find(`#${mockCheckboxData.id}`).prop('id')).toEqual(
			mockCheckboxData.id
		)

		expect(
			checkboxWrapper.find(`#${mockCheckboxData.id}`).prop('checked')
		).toEqual(mockCheckboxData.isChecked)
	})

	it('checks the DOM events', () => {
		checkboxWrapper
			.find(`#${mockCheckboxData.id}`)
			.simulate('change', { target: { checked: false } })

		expect(handleChangeStub).toHaveBeenCalledTimes(1)
	})

	it('matches the snaphot', () => {
		expect(checkboxWrapper).toMatchSnapshot()
	})
})
