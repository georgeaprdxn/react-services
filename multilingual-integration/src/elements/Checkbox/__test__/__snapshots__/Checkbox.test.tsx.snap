// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox element tests matches the snaphot 1`] = `
<styled.div>
  <styled.input
    checked={true}
    id="some-id"
    name="Some name"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "target": Object {
                "checked": false,
              },
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    type="checkbox"
  />
  <styled.label
    htmlFor="some-id"
    title="Some title"
  >
    Some title
  </styled.label>
</styled.div>
`;
